README.md
COLLEC-SCIENCE
============
Collec est un logiciel destiné à gérer les collections d'échantillons prélevés sur le terrain.

Écrit en PHP, il fonctionne avec une base de données Postgresql. Il est bâti autour de la notion d'objets, qui sont identifiés par un numéro unique. Un objet peut être de deux types : soit un container (aussi bien un site, un bâtiment, une pièce, un congélateur, une caisse...) qu'un échantillon. 
Un type d'échantillon peut être rattaché à un type de container, quand les deux notions se superposent (le flacon contenant le résultat d'une pêche est à la fois un container et l'échantillon lui-même).
Un objet peut se voir attacher plusieurs identifiants métiers différents, des événements, ou des réservations.
Un échantillon peut être subdivisé en d'autres échantillons (du même type ou non). Il peut contenir plusieurs éléments identiques (notion de sous-échantillonnage), comme des écailles de poisson indifférenciées.
Un échantillon est obligatoirement rattaché à un projet. Les droits de modification sont attribués au niveau du projet.

Fonctionnalités principales
---------------------------
- Entrée/sortie du stock de tout objet (un container peut être placé dans un autre container, comme une boite dans une armoire, une armoire dans une pièce, etc)
- possibilité de générer des étiquettes avec ou sans QRCODE
- gestion d'événements pour tout objet
- réservation de tout objet
- lecture par scanner (douchette) des QRCODE, soit objet par objet, soit en mode batch (lecture multiple, puis intégration des mouvements en une seule opération)
- lecture individuelle des QRCODES par tablette ou smartphone (testé, mais pas très pratique pour des raisons de performance)
- ajout de photos ou de pièces jointes à tout objet

Sécurité
--------
- logiciel homologué à Irstea, résistance à des attaques opportunistes selon la nomenclature de l'OWASP (projet ASVS), mais probablement capable de répondre aux besoins du niveau standard
- identification possible selon plusieurs modalités : base de comptes interne, annuaire ldap, ldap - base de données (identification mixte), via serveur CAS, ou par délégation à un serveur proxy d'identification, comme LemonLDAP, par exemple
- gestion des droits pouvant s'appuyer sur les groupes d'un annuaire LDAP

Licence
-------
Logiciel diffusé sous licence AGPL

Copyright
---------
La version 1.0 a été déposée auprès de l'Agence de Protection des Programmes sous le numéro IDDN.FR.001.470013.000.S.C.2016.000.31500


param/news.txt
News
===========
Version 2.0.1 du 25 mai 2018
--------------------------
Corrections de bugs :
- les listes de choix étaient illisibles sur de petits terminaux
- quand l'interface était positionnée en anglais, la création d'un nouvel utilisateur générait une erreur de gestion de dates
- la génération du fichier pour imprimer des étiquettes (export CSV) ne traitait pas correctement les métadonnées, qui étaient mal positionnées dans les colonnes
- le script col_alter_1.2-1.2.3.sql pouvait générer des erreurs dans certains cas de figure
- si une requête de recherche des échantillons déclenchait une erreur de la base de données, l'écran restait blanc sans pouvoir savoir ce qu'il se passait

Évolutions :
- ajout d'un mode d'affichage restreint de la liste des échantillons, soit pour des usages spécifiques, soit pour les petits écrans (seuls quelques colonnes sont affichées). La bascule s'opère via un bouton
- ajout d'une précision pour gérer correctement les droits dans col_alter_1.2.3-2.0.sql
- l'export des échantillons vers une autre base crée des colonnes dédiées pour chaque métadonnée, préfixée par md_

Version 2.0 du 4 mai 2018
--------------------------
Évolutions :
- mise à jour du framework et de la base de données
- possibilité de générer l'étiquette de l'échantillon en cours de visualisation
- lors de la création d'un nouvel échantillon, les informations présentes dans l'étiquette pré-existante (format Json) sont récupérées automatiquement après son scannage
- ajout d'un filtre pour supprimer les caractères parasites lors de la lecture des codes barres 1D de type EAN128
- lors de la saisie d'un mouvement à partir d'un petit terminal, ajout du dernier statut de l'objet scanné (présent/absent) et d'une temporisation pour limiter les interrogations du serveur
- ajout d'auto-complétion dans quelques boites déroulantes
- en recherche d'échantillon, il est maintenant possible de rechercher jusqu'à 3 critères de métadonnées, qui fonctionnent soit en mode "et" (champs différents), soit en mode "ou" (champs identiques). Pour des questions de performance, les champs utilisables pour la recherche doivent avoir été décrits comme tels (création d'index)
- le terme "projet" a été renommé en "collection", tant dans l'application que dans la base de données
- meilleure définition des dates associées aux échantillons : création dans la base de données, prélèvement, et ajout d'une date d'expiration
- il est maintenant possible de rechercher un échantillon par date (de création, de prélèvement, d'expiration)
- il est possible de générer automatiquement l'identifiant métier à partir des informations saisies (code à définir par type d'échantillon)
- les informations liées aux lieux de prélèvement ont été complétées par des coordonnées GPS et un code. Les coordonnées sont reportées dans l'échantillon si celui-ci n'en dispose pas. 
- il est possible de définir des lieux de prélèvement qui sont uniquement attachés à une collection
- lors de l'importation externe, les champs de métadonnées peuvent être indiqués dans des colonnes séparées, préfixées par "md_"
- l'importation externe fonctionne maintenant en mode "remplace ou crée", ce qui permet de mettre à jour des échantillons manipulés dans des bases externes
- lors de l'importation externe, la filiation d'un échantillon est maintenant conservée (si le parent a été créé au préalable)
- en génération des étiquettes, il est maintenant possible d'imprimer le champ dbuid-origin
- il est possible d'accéder directement à un échantillon à partir de son lien d'accès
- de nouveaux scripts sont disponibles pour faciliter l'installation d'un nouveau serveur ou mettre à jour une base de données

Corrections de bugs :
- en recherche d'échantillons, correction de la recherche sur les métadonnées 
- en création des étiquettes, un bug empêchait l'affichage des champs de métadonnées
- l'importation de métadonnées externes ne récupérait pas correctement le champ Json
- correction du message de réinitialisation du mot de passe
- harmonisation des libellés concernant les conteneurs
- ajout de contrôles supplémentaires lors de l'importation externe, avec test de cohérence des dates
- la sortie des échantillons, depuis le détail d'un échantillon, n'aboutissait pas
- en création de mouvements de masse, les échantillons externes n'étaient pas reconnus
- lors de la procédure d'importation externe, l'appariement était fait en tenant compte de la casse, ce qui a été supprimé
- en création de mouvements de masse, le logiciel ne savait pas gérer correctement le fait que le même container était scanné plusieurs fois
- le lien "retour à la liste" ne ramenait pas toujours à la liste d'origine, si l'utilisateur avait basculé de la consultation d'un échantillon à un conteneur et vice-versa
- le basculement entre le français et l'anglais posait des problèmes de gestion de la date

Version 1.2.4 du 5/12/2017
---------------------------
Correction de bugs :
- la création d'un compte de connexion ne fonctionnait plus : le mot de passe n'était jamais enregistré

Version 1.2.3 du 22/11/2017
---------------------------
Évolutions :
- l'import de masse accepte maintenant l'UID de l'échantillon parent
- le droit "import" est créé, il permet d'autoriser un utilisateur à importer les informations via des fichiers csv (import de masse ou d'une autre instance)
- l'écran d'import de masse a été remanié, en affichant les boîtes de dialogue en haut d'écran
- les performances lors de la recherche d'un échantillon sont optimisées par l'utilisation d'un nouveau type d'index

Corrections de bugs :
- lors de l'import de masse, les métadonnées n'étaient pas correctement reconnues (bug introduit en version 1.2)
- la suppression d'un login ou d'un compte dans la gestion des droits a été améliorée, elle supprime également les dépendances croisées (login, compte déclaré, présence dans les groupes)

locales/fr.php
<?php
$_SESSION["FORMATDATE"]="fr";
$_SESSION["MASKDATELONG"] = "d/m/Y H:i:s";
$_SESSION["MASKDATE"] = "d/m/Y";
$_SESSION["MASKDATEEXPORT"] = 'd-m-Y';

$LANG= array();
$LANG["date"]["locale"] = "fr";
$LANG["date"]["formatdate"] = "DD/MM/YYYY";
$LANG["date"]["formatdatetime"] = "DD/MM/YYYY HH:mm:ss";
$LANG["date"]["formatdatecourt"] = "dd/mm/yy";
/*
 * menu
 */
$LANG["menu"][0] = "Gestion";
$LANG["menu"][1] = "Opérations de gestion";
$LANG["menu"][2] = "Liste des comptes locaux";
$LANG["menu"][3] = "Liste des logins - identification via la base de données";
$LANG["menu"][4] = "Administration";
$LANG["menu"][5] = "Administration de l'application";
$LANG["menu"][6] = "Déconnexion";
$LANG["menu"][7] = "Déconnexion de l'application";
$LANG["menu"][8] = "A propos";
$LANG["menu"][9] = "A propos de prototypePHP";
$LANG["menu"]["help"] = "?";
$LANG["menu"]["helpTitle"] = "Aide";

$LANG["menu"][40] = "Nouveautés";
$LANG["menu"][41] = "Quoi de neuf dans l'application ?";
$LANG["menu"]["phpinfo"] = "PHP info";
$LANG["menu"]["phpinfotitle"] = "configuration générale du serveur PHP";
$LANG["menu"]["fr"] = "français";
$LANG["menu"]["frtitle"] = "Libellés en français";
$LANG["menu"]["en"] = "english";
$LANG["menu"]["entitle"] = "labels in english";
$LANG["menu"]["pwchange"] = "Modifier le mot de passe";
$LANG["menu"]["pwchangetitle"] = "Modifiez votre mot de passe (identification locale uniquement)";

/*
 * Gestion des droits nouvelle formule
 */
$LANG["menu"]["appliliste"] = "ACL - droits";
$LANG["menu"]["applilistetitle"] = "applications et droits gérés";
$LANG["menu"]["aclloginliste"] = "ACL - logins";
$LANG["menu"]["aclloginlistetitle"] = "Logins déclarés dans le module de gestion des droits";
$LANG["menu"]["groupliste"] = "ACL - groupes de logins";
$LANG["menu"]["grouplistetitle"] = "Groupes de logins et logins rattachés aux groupes";

$LANG["menu"]["example1"] = "Example";
$LANG["menu"]["example2"] = "It's an example !";
$LANG["menu"]["connexionvalue"] = "Connexion";
$LANG["menu"]['connexiontitle'] = "S'identifier dans l'application";

/*
 * Entrees du menu pour l'application
 */
$LANG["menu"][50] = "Paramètres";
$LANG["menu"][51] = "Paramètres";
$LANG["menu"][52] = "Type événement";
$LANG["menu"][53] = "Types d'événement";
$LANG["menu"][54] = "Familles de conteneurs";
$LANG["menu"][55] = "Familles de conteneurs (armoires, éprouvettes...)";
$LANG["menu"][56] = "Conditions de stockage";
$LANG["menu"][57] = "Liste des conditions de stockage";
$LANG["menu"][58] = "Types de conteneurs";
$LANG["menu"][59] = "Liste des types de conteneurs";
$LANG["menu"][60] = "Conteneurs";
$LANG["menu"][61] = "Liste des conteneurs";
$LANG["menu"][62] = "Statuts des objets";
$LANG["menu"][63] = "Liste des statuts possibles pour les objets";
$LANG["menu"][64] = "Collections";
$LANG["menu"][65] = "Liste des collections";
$LANG["menu"][66] = "Type d'échantillons";
$LANG["menu"][67] = "Liste des types d'échantillons gérés";
$LANG["menu"][68] = "Échantillons";
$LANG["menu"][69] = "Liste des échantillons";
$LANG["menu"][70] = "Mouvements";
$LANG["menu"][71] = "Création de mouvements dans le stock";
$LANG["menu"][72] = "Entrer un échantillon";
$LANG["menu"][73] = "Entrée rapide d'un échantillon dans le stock";
$LANG["menu"][74] = "Sortir un échantillon";
$LANG["menu"][75] = "Sortie rapide d'un échantillon du stock";
$LANG["menu"][76] = "Objets";
$LANG["menu"][77] = "Opérations sur les échantillons ou les containers";
$LANG["menu"][78] = "Import de masse";
$LANG["menu"][79] = "Import de plusieurs échantillons ou containers";
$LANG["menu"][80] = "Étiquettes";
$LANG["menu"][81] = "Modèles d'étiquettes";
$LANG["menu"][82] = "Protocoles";
$LANG["menu"][83] = "Liste des protocoles";
$LANG["menu"][85] = "Opérations";
$LANG["menu"][86] = "Liste des opérations rattachées à un protocole";
$LANG["menu"][87] = "Sous-échantillonnage";
$LANG["menu"][88] = "Types de sous-échantillonnage";
$LANG["menu"][89] = "Types d'identifiants";
$LANG["menu"][90] = "Liste des types d'identifiants rattachables à un objet";
$LANG["menu"][91] = "Motifs de déstockage";
$LANG["menu"][92] = "Motifs de sortie du stock";
$LANG["menu"][93] = "Entrée/sortie par lots";
$LANG["menu"][94] = "Entrée / sortie par scan multiple de QRCODE";
$LANG["menu"][95] = "Lieux de prélèvement";
$LANG["menu"][96] = "Liste des lieux de prélèvement des échantillons";
$LANG["menu"][97] = "Imprimantes";
$LANG["menu"][98] = "Liste des imprimantes";
$LANG["menu"][99] = "Métadonnées";
$LANG["menu"][100] = "Modèles de métadonnées";
$LANG["menu"][101] = "Import d'échantillons externes";
$LANG["menu"][102] = "Import d'un lot d'échantillons provenant d'une base externe";
$LANG["menu"][103] = "Paramètres de l'application";
$LANG["menu"][104] = "Liste des paramètres pérennes de l'application";
$LANG["menu"][105] = "Liste des mouvements";
$LANG["menu"][106] = "Recherche des mouvements sur une période donnée";
$LANG["menu"][107] = "Lexique";
$LANG["menu"][108] = "Lexique des termes utilisés dans le logiciel";
$LANG["menu"][109] = "Mouvements petit terminal";
$LANG["menu"][110] = "Lecture d'objets et génération de mouvements sur petits terminaux";
$LANG["menu"][111] = "";
$LANG["menu"][112] = "";




$LANG["message"][0] = "Bienvenue"; 
$LANG["message"][1] = "GESTION DES ÉCHANTILLONS";
$LANG["message"][2] = "Module administration";
$LANG["message"][3] = "Module gestion";
$LANG["message"][4] = "Suppression effectuée";
$LANG["message"][5] = "Enregistrement effectué";
$LANG["message"][6] = "Vidage effectué";
$LANG["message"][7] = "Vous êtes maintenant déconnecté";
$LANG["message"][8] = "Connexion";
$LANG["message"][9] = "Vous n'avez pas le droit d'accéder à cette fonction";
$LANG["message"][10] = "Identification réussie !";
$LANG["message"][11] = "Identification refusée";
$LANG["message"][12] = "Problème lors de la mise en fichier...";
$LANG["message"][13] = "Problème lors de la suppression";
$LANG["message"][14] = "Vous ne pouvez pas accéder directement à cette page";
$LANG["message"][15] = "oui";
$LANG["message"][16] = "non";
$LANG["message"][17] = "Attention :";
$LANG["message"][18] = "Confirmez-vous l'opération ?";
$LANG["message"][19] = "Valider";
$LANG["message"][20] = "Supprimer";
$LANG["message"][21] = "Rechercher";
$LANG["message"][22] = "Echec de connexion à la base de données principale";
$LANG["message"][23] = "Copyright (c) 2016-2018 - Tous droits réservés. Auteur : Éric Quinton, pour IRSTEA - Logiciel diffusé sous licence AGPL";
$LANG["message"][24] = "Logiciel diffusé sous licence AGPL.";
$LANG["message"][25] = "Pour tout problème : ";
$LANG["message"][26] = "Vous n'avez pas accès à cette fonction. Veuillez vous identifier, le cas échéant...";
$LANG["message"][27] = "Vous n'êtes pas identifié.";
$LANG["message"][28] = "L'accès au module que vous demandez ne respecte pas la cinématique de l'application. 
		Peut-être avez-vous essayé de rafraîchir la page ?";
$LANG["message"][29] = "Echec de connexion à la base de données de gestion des droits (GACL)";
$LANG["message"][30] = "Champ obligatoire";
$LANG["message"][31] = "Confirmez-vous la suppression ?";
$LANG["message"][32] = "Mode développement - base de données";
$LANG["message"][33] = "Déconnexion";
$LANG["message"][34] = "Valeur numérique...";
$LANG["message"][35] = "Le module demandé n'existe pas";
$LANG["message"][36] = "Donnée obligatoire";
$LANG["message"][37] = "Erreur de lecture des informations dans la base de données";
$LANG["message"][38] = "Erreur d'écriture dans le fichier de traces";
$LANG["message"][39] = "Les mots de passe ne sont pas identiques";
$LANG["message"][40] = "Le mot de passe est trop court (minimum : 10 caractères)";
$LANG["message"][41] = "Le mot de passe n'est pas assez complexe (mixez 3 jeux de caractères parmi les minuscules, majuscules, chiffres et signes de ponctuation)";
$LANG["message"][43] = "Suppression du groupe impossible : d'autres groupes lui sont rattachés";
$LANG["message"][44] = "Vous avez été déconnecté, votre session était ouverte depuis trop longtemps";
$LANG["message"][45] = "Problème dans les données fournies : l'encodage des caractères n'est pas celui attendu";
$LANG["message"][46] = "La base de données n'est pas dans la version attendue (%). Version actuelle : ";
$LANG["message"][47] = "Module precedent enregistré : ";
$LANG["message"][48] = "L'identification par jeton n'a pas abouti";
$LANG["message"][49] = "";
$LANG["message"][50] = "";

$LANG["message"]["yes"] = "oui";
$LANG["message"]["no"] = "non";

$LANG["login"][0] = "Login";
$LANG["login"][1] = "Mot de passe";
$LANG["login"][2] = "Veuillez utiliser votre login du domaine pour vous identifier";
$LANG["login"][5] = "Nouveau login";
$LANG["login"][6] = "login";
$LANG["login"][7] = "Nom - prénom";
$LANG["login"][8] = "Mail";
$LANG["login"][9] = "Nom";
$LANG["login"][10] = "Prénom";
$LANG["login"][11] = "Date";
$LANG["login"][12] = "Répétez le mot de passe";
$LANG["login"][13] = "actif";
$LANG["login"][14] = "Le mot de passe a déjà été utilisé";
$LANG["login"][15] = "Le mot de passe n'est pas assez complexe";
$LANG["login"][16] = "Le mot de passe est trop court";
$LANG["login"][17] = "Le mot de passe n'est pas identique dans les deux zones";
$LANG["login"][18] = "Le mode d'identification utilisé pour votre compte n'autorise pas la modification du mot de passe depuis cette application";
$LANG["login"][19] = "L'ancien mot de passe est incorrect";
$LANG["login"][20] = "Le mot de passe a été modifié";
$LANG["login"][21] = "Générez un mot de passe aléatoire";
$LANG["login"][22] = "Générez";
$LANG["login"][23] = "Ancien mot de passe";
$LANG["login"][24] = "Nouveau mot de passe";
$LANG["login"][25] = "Le mot de passe doit avoir une longueur minimale de 10 caractères<br>
Il doit comprendre au minimum 3 types de caractères différents<br>
(minuscule, majuscule, chiffre, ponctuation et autre symboles)";
$LANG["login"][26] = "Droits attribués";
$LANG["login"][27] ="Modification d'un login (module de gestion des droits)";
$LANG["login"][28] ="Retour à la liste des logins";
$LANG["login"][29] ="Nom de l'utilisateur";
$LANG["login"][30] ="Login utilisé";
$LANG["login"][31] ="Modifier le mot de passe";
$LANG["login"][33] ="Liste des groupes de logins";
$LANG["login"][34] ="Nouveau groupe racine...";
$LANG["login"][35] ="Nom du groupe";
$LANG["login"][36] ="Nombre de logins déclarés";
$LANG["login"][37] ="Rajouter un groupe fils";
$LANG["login"][38] ="Modification d'un groupe et rattachement des logins";
$LANG["login"][39] ="Retour à la liste des groupes";
$LANG["login"][40] ="Logins rattachés";
$LANG["login"][41] ="Modification du droit d'une application (module de gestion des droits)";
$LANG["login"][42] ="Retour à la liste des applications";
$LANG["login"][43] ="Retour à";
$LANG["login"][44] ="Nom du droit utilisé dans l'application";
$LANG["login"][45] ="Groupes disposant du droit";
$LANG["login"][46] ="Conserver la connexion pendant";
$LANG["login"][47] = "heures";
$LANG["login"][48] = "Dernière connexion le :datelog depuis l'adresse IP :iplog. Si ce n'était pas vous, modifiez votre mot de passe ou contactez l'administrateur de l'application.";
$LANG["login"][49] = "L'accès au module demandé nécessite une ré-identification. Veuillez saisir votre login et votre mot de passe";
$LANG["login"][50] = "Echec de modification du mot de passe pour une raison inconnue. Si le problème persiste, contactez l'assistance";
$LANG["login"][51] = "La modification du mot de passe a échoué";
$LANG["login"][52] = "Le jeton fourni n'est pas valide";
$LANG["login"][53] = "<html><body>:prenom :nom,<br>
<br>Vous avez demandé à réinitialiser votre mot de passe pour l'application :appli. Si ce n'était pas le cas, contactez l'administrateur de l'application.
<br>Pour réinitaliser votre mot de passe, recopiez le lien suivant dans votre navigateur :<br><a href=':adresse'>Réinitialisez votre mot de passe</a>
<br>Ne répondez pas à ce mail, qui est généré automatiquement</body></html>";
$LANG["login"][54] = ":appli - réinitialisation du mot de passe";
$LANG["login"][55] = "Un mail vient de vous être envoyé. Veuillez copier le lien transmis dans votre navigateur pour pouvoir créer un nouveau mot de passe";
$LANG["login"][56] = "Impossible d'envoyer le mail";
$LANG["login"][57] = "Entrez votre mail";
$LANG["login"][58] = "Mot de passe perdu - lancement de la procédure de récupération";
$LANG["login"][59] = "Perdu votre mot de passe ?";
$LANG["login"][60] = "Compte utilisé pour service web";
$LANG["login"][61] = "Jeton d'identification du service web";
$LANG["login"][62] = "Réinitialiser...";
$LANG["login"][63] = "Copier dans le presse-papier";

$LANG["ObjetBDDError"][0] = "Le champ ";
$LANG["ObjetBDDError"][1] = " n'est pas numerique.";
$LANG["ObjetBDDError"][2] = " est trop grand. Longueur maximale autorisée : ";
$LANG["ObjetBDDError"][3] = " Valeur saisie : ";
$LANG["ObjetBDDError"][4] = " caractères";
$LANG["ObjetBDDError"][5] = "Le contenu du champ ";
$LANG["ObjetBDDError"][6] = " ne correspond pas au format attendu. Masque autorisé : ";
$LANG["ObjetBDDError"][7] = " est obligatoire, mais n'a pas été renseigné.";

$LANG["appli"][0] = "Nouveau...";
$LANG["appli"][1] = "Retour à la liste";
$LANG["appli"][2] = "Choisissez...";
$LANG["appli"][3] = "Le conteneur n'a pas été renseigné";
$LANG["appli"][4] = "Vous ne disposez pas des droits pour modifier cet échantillon";
$LANG["appli"][5] = "Les contrôles de cohérence ne permettent pas d'enregister le mouvement demandé. ";
$LANG["appli"][6] = "Impossible d'enregistrer le mouvement";
$LANG["appli"][7] = "Mouvement enregistré";

?>
